crates/tessera-backbone/src/domain/secret/mod.rs

```diff
@@ -265,20 +265,79 @@ impl From<(secret_metadata::Model, Vec<applied_policy::Model>, Vec<u8>)> for Sec
 
 pub(crate) struct Path {
     pub path: String,
+    deleted: bool,
+}
+
+impl Path {
+    #[cfg(test)]
+    pub(crate) fn new(path: String) -> Self {
+        Self { path, deleted: false }
+    }
+
+    pub(crate) fn delete(&mut self) {
+        self.deleted = true
+    }
+
+    async fn ensure_child_path_not_exists(&self, transaction: &DatabaseTransaction) -> Result<()> {
+        if path::Entity::find()
+            .filter(path::Column::Path.like(&self.path))
+            .filter(path::Column::Path.ne(&self.path))
+            .count(transaction)
+            .await?
+            > 0
+        {
+            return Err(Error::PathIsInUse { entered_path: self.path.to_owned().to_owned() });
+        }
+
+        Ok(())
+    }
+
+    async fn ensure_child_secret_not_exists(&self, transaction: &DatabaseTransaction) -> Result<()> {
+        if secret_metadata::Entity::find()
+            .filter(secret_metadata::Column::Path.like(&self.path))
+            .count(transaction)
+            .await?
+            > 0
+        {
+            return Err(Error::PathIsInUse { entered_path: self.path.to_owned().to_owned() });
```

Comment:
The double `.to_owned()` calls appear redundant here. I think a single `.to_owned()` should be sufficient to create an owned copy of the path.   

Alternatively, if the type implements the `ToString` trait, using `.to_string()` would be more idiomatic. Here are the suggested alternatives:  

```rust  
// Option 1: Single to_owned()  
return Err(Error::PathIsInUse { entered_path: self.path.to_owned() });  

// Option 2: Using to_string() if ToString is implemented  
return Err(Error::PathIsInUse { entered_path: self.path.to_string() });